forte/processors/base/base_processor.py

```diff
@@ -38,11 +39,34 @@ def __init__(self):
         super().__init__()
         self.selector = DummySelector()
 
+    def record(self, record_meta: Dict):
+        r"""Method to add output type record of current processor
+        to Meta attribute record dictionary.
+        """
+        pass
+
+    @classmethod
+    def expected_type(cls) -> Dict:
+        r"""Method to add expected type for current processor input which
+        would be checked before running the processor if enforce_consistency()
+        was called for the pipeline.
+        """
+        return {}
+
     def process(self, input_pack: PackType):
+        if self._check_type_consistency:
+            expected_types = self.expected_type()
+            # check if expected types are in input pack
+            for expected_t in expected_types:
```

Comment:
Here only the type is checked, but the attributes are not checked. Can you check for those as well?